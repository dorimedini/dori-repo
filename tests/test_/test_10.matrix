// Test initializations
{
    int a;
    print a;

    int b = 5;
    print b;

    int c = -6;
    print c;

    int multiple = 10, vars = 11, on = 12, same = 13, line = 14;
    print multiple;
    print vars;
    print on;
    print same;
    print line;

    int variable_name_with_all_allowed_characters_12345 = 20;
    print variable_name_with_all_allowed_characters_12345;

    matrix m1(2, 3);
    print m1;

    matrix m2(1, 1) = [1];
    print m2;

    matrix m3(1, 1) = [c];
    print m3;

    matrix m4(2, 2) = [1, 2; 4, 5];
    print m4;

    matrix m5(2, 3) = [1+5, 2-4, 3*3; 4/3, 5*(3-1), 6];
    print m5;

    matrix m6(3, 2) = [a, b; c, a; b, c];
    print m6;

    matrix m7(2, 2) = [a, b; c, a] + [10, 11, 12; 13, 14, 15] * transpose([20, 21, 22; 23, 24, 25]);
    print m7;

    matrix mult(1, 2) = [10, 11], matrices(2, 1) = [12; 13];
    print mult;
    print matrices;
}
