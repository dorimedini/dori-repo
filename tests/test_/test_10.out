0
5
-6
10
11
12
13
14
20
[0,0,0;0,0,0]
[1]
[-6]
[1,2;4,5]
[6,-2,9;1,10,6]
[0,5;-6,0;5,-6]
[695,799;878,1010]
[10,11]
[12;13]
